libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/LiveRideOverlay.java

```diff
@@ -60,7 +80,35 @@ public LiveRideOverlay(final Activity context, final Locator locator) {
 
     final Rect bounds = new Rect();
     largeTextBrush_.getTextBounds("0.0", 0, 3, bounds); // Measure the text
-    lineHeight_ = bounds.height();
+    largeLineHeight_ = bounds.height();
+    midTextBrush_.getTextBounds("0.0", 0, 3, bounds);
+    midLineHeight_ = bounds.height();
+    smallTextBrush_.getTextBounds("0.0", 0, 3, bounds);
+    smallLineHeight_ = bounds.height();
+
+    remainingText = context.getString(R.string.remaining); //"Remaining";
+    ETAText = " " + context.getString(R.string.ETA); //" / ETA";
+    remainingWidth = (int)smallTextBrush_.measureText(remainingText + ":");
+    showETA = CycleStreetsPreferences.showETA();
+    showRemainingTime = CycleStreetsPreferences.showRemainingTime();
+    showTime = showETA || showRemainingTime;
+    if (!showTime) {
+      height = largeLineHeight_ + offset_*2;
+      // If time not shown, no need to reduce height of bottom box after 1st seg
+      heightNoText = height;
+    }
+    else {
+      height = smallLineHeight_ + midLineHeight_ * 2 + offset_ * 4;
+      // After 1st seg, text will be removed from bottom box and height reduced
+      heightNoText = height - smallLineHeight_;
+    }
+    TileSource.setBoxHeight(height);  // Use this height to shift map attribution in ControllerOverlay
+    if (showTime) {
+      titleVerticalPosition = offset_ + midLineHeight_ * 2;
+    }
+    else {
+      titleVerticalPosition = (largeLineHeight_ - smallLineHeight_ - midLineHeight_ - offset_) / 2 + midLineHeight_ + offset_;
+    }
```

Comment:
This new block of code is too much to put into the constructor.  Please contextualise.